sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -800,15 +800,40 @@ protected virtual TimeSpan CalculateLinkAuthorizationRefreshInterval(DateTime ex
         }
 
         /// <summary>
-        ///   Performs the actions needed to open a generic AMQP object, such
+        ///   Performs the actions needed to open an AMQP object, such
         ///   as a session or link for use.
         /// </summary>
         ///
         /// <param name="target">The target AMQP object to open.</param>
         /// <param name="timeout">The timeout to apply when opening the link.</param>
         ///
-        protected virtual Task OpenAmqpObjectAsync(AmqpObject target,
-                                                   TimeSpan timeout) => target.OpenAsync(timeout);
+        protected virtual async Task OpenAmqpObjectAsync(AmqpObject target,
```

Comment:
Unfortunately, the types from the AMQP library do not support mocking which makes these changes nearly impossible to test.  I originally had `OpenAmqpLinkObjectAsync` and `OpenRequestResponseAmqpLink` as `protected virtual` method with rudimentary coverage around the switch, but ultimately decided the minor benefits of that test were outweighed by the efficiency improvement of avoiding the `virtual` calls and hoisting to a `static` member.